### DISCUSSION TITLE
[SSR] Running Lit SSR on Stackblitz

### DISCUSSION THREAD

**Discussion by thescientist13** [lit/lit#2087] [General]
## Overview
Just wanted to track it here as well for anybody curious or interested in seeing if Lit's SSR can run on [Stackblitz and Web Containers](https://blog.stackblitz.com/posts/introducing-webcontainers/).  (If you don't know about Web Containers I highly recommend checking them out!)

I created a [sample repo anyone can use for testing](https://github.com/thescientist13/lit-ssr-demo) and [opened an issue](https://github.com/stackblitz/webcontainer-core/issues/307) after observing a couple issues with running that repo on Stackblitz, and will quickly summarize them here.

## Global Scope
If you run `npm run serve:global`, Stackblitz will show this error and fail to start the server.
```sh
❯ npm run serve:global
$ node ./src/global/server.js
Error: document is not defined
    at async Promise.all (index 0)
    at async _0x31ad20._evaluate (https://bnmijzjjj--github.jw.staticblitz.com/blitz.8ffc536d8181e8b30c228dec423d18afae547c40.js:11:196274)
    at async Promise.all (index 6)
```
<img width="1669" alt="Screen Shot 2021-08-24 at 10 00 36 AM" src="https://user-images.githubusercontent.com/895923/130630647-10f1f8d6-23d4-4c5e-8298-40bc8cea770d.png">

## VM Module
If you run `npm run serve:vm`, Stackblitz will show successful server logs in the console, but fail to open the server in the browser pane.
<img width="1677" alt="Screen Shot 2021-08-24 at 10 01 23 AM" src="https://user-images.githubusercontent.com/895923/130630864-0a0b4a29-19f9-4f6d-9e21-b41c8dc0550b.png">

**Comment by justinfagnani** [comment]
This is very cool, thanks for sharing!

We'll be getting back to SSR work once the Lit 2 release is out. I'm excited to look at what everyone in the community's been up to!

**Comment by thescientist13** [comment]
An update from the Stackblitz team, looks like `--experimental-vm-modules` has landed!
https://github.com/stackblitz/webcontainer-core/issues/307#issuecomment-1100182561
